sources/packages/backend/apps/api/src/services/application-restriction-bypass/application-restriction-bypass.service.ts

```diff
@@ -43,10 +67,382 @@ export class ApplicationRestrictionBypassService extends RecordDataModelService<
+   * @returns application restriction bypass.
+   */
+  async getApplicationRestrictionBypass(
+    id: number,
+  ): Promise<ApplicationRestrictionBypass> {
+    return this.applicationRestrictionBypassRepo.findOne({
+      select: {
+        id: true,
+        isActive: true,
+        studentRestriction: {
+          id: true,
+          restriction: {
+            restrictionCode: true,
+          },
+        },
+        creationNote: {
+          description: true,
+        },
+        removalNote: {
+          description: true,
+        },
+        bypassCreatedBy: {
+          firstName: true,
+          lastName: true,
+        },
+        bypassCreatedDate: true,
+        bypassRemovedBy: {
+          firstName: true,
+          lastName: true,
+        },
+        bypassRemovedDate: true,
+        bypassBehavior: true,
+      },
+      relations: {
+        studentRestriction: { restriction: true },
+        creationNote: true,
+        removalNote: true,
+        bypassCreatedBy: true,
+        bypassRemovedBy: true,
+      },
+      where: {
+        id,
+      },
+    });
+  }
+
+  /**
+   * Gets all available student restrictions to bypass for a given application.
+   * @param applicationId application id.
+   * @returns list of application restriction bypass options.
+   */
+  async getAvailableStudentRestrictionsToBypass(
+    applicationId: number,
+  ): Promise<StudentRestriction[]> {
+    const applicationRestrictionBypasses =
+      await this.applicationRestrictionBypassRepo.find({
+        select: {
+          studentRestriction: {
+            id: true,
+          },
+        },
+        relations: {
+          studentRestriction: true,
+        },
+        where: {
+          application: {
+            id: applicationId,
+          },
+          isActive: true,
+        },
+      });
+    const studentRestrictionIds = applicationRestrictionBypasses.map(
+      (applicationRestrictionBypass) =>
+        applicationRestrictionBypass.studentRestriction.id,
+    );
+
+    const application = await this.applicationRepo.findOne({
+      select: {
+        student: {
+          id: true,
+        },
+        currentAssessment: {
+          id: true,
+          offering: {
+            offeringIntensity: true,
+          },
+        },
+      },
+      relations: {
+        student: true,
+        currentAssessment: {
+          offering: true,
+        },
+      },
+      where: {
+        id: applicationId,
+      },
+    });
+
+    const query = this.studentRestrictionRepo
+      .createQueryBuilder("studentRestriction")
+      .select([
+        "studentRestriction.id",
+        "restriction.restrictionCode",
+        "studentRestriction.createdAt",
+      ])
+      .innerJoin("studentRestriction.restriction", "restriction")
+      .andWhere("studentRestriction.student.id = :studentId", {
+        studentId: application.student.id,
+      })
+      .andWhere("studentRestriction.id NOT IN (:...studentRestrictionIds)", {
```

Comment:
Is there a check needed for the case where the `studentRestrictionIds` will be empty?